File: frontend/materials/stores/projectStore.ts
import { create } from 'zustand';
import { produce } from 'immer';
import { PageVariable } from '@materials/types';


export interface ProjectState {
  variables: PageVariable[];
  variableData: { [key: string]: any };
}

export interface ProjectAction {
  setVariables: (payload: PageVariable[]) => void;
  setVariableData: (payload: any) => void;
}

export const useProjectStore = create<ProjectState & ProjectAction>((set, get) => ({
  // 项目变量定义列表
  variables: [],
  // 项目变量数据
  variableData: {},
  // 初始化变量列表
  setVariables(payload: PageVariable[]) {
    set(
      produce((state) => {
        state.variables = payload;
      }),
    );
  },
  // 设置变量数据
  setVariableData({ name, value }: any) {
    set(
      produce((state) => {
        state.variableData[name] = value;
      }),
    );
  },
}));

File: backend/code-vue/src/templates/stores/project_store.rs
pub const PROJECT_STORE_INDEX: &str = r#"
import { defineStore } from 'pinia';
import { reactive } from 'vue';
import type { PageVariable } from '@/types';

export interface ProjectState {
  variables: PageVariable[];
  variableData: { [key: string]: any };
}

export const useProjectStore = defineStore('projectStore', () => {
  const projectState = reactive<ProjectState>({
    variables: [],
    variableData: {},
  });

  // 保存变量信息
  const setVariables = (variables: PageVariable[]) => {
    projectState.variables = variables;
  };

  // 保存变量数据
  const setVariableData = ({ name, value }: any) => {
    projectState.variableData[name] = value;
  };
  return {
    projectState,
    setVariables,
    setVariableData,
  };
})
"#;